### PR TITLE
doc: Docstring checker to fix returns section markup

### LINKED ISSUE #3397: Fix documentation markup of "returns" section
In many documentation pages, markup of "returns" section is broken. This needs to be fixed.

a

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #3397.

Fixed corrupted "Returns" section.

I introduced the docstring checker which is triggered by sphinx autodoc.
It can perform arbitrary check if we implement a rule.

### CODE REVIEW

**Comment by niboshi** [comment]
Rebased to master.

File: docs/source/_docstring_check.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        self.name = name
+        self.obj = obj
+        self.options = options
+        self.lines = lines
+
+        self.iline = 0
+
+    def nextline(self):
+        if self.iline >= len(self.lines):
+            raise StopIteration()
```
[the remark this answers is not in the document]
Comment (niboshi) [reply docs/source/_docstring_check.py]: Fixed.

File: docs/source/_docstring_check.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import math
+
+
+def check(app, what, name, obj, options, lines):
+    ctx = DocstringCheckContext(app, what, name, obj, options, lines)
```
[the remark this answers is not in the document]
Comment (niboshi) [reply docs/source/_docstring_check.py]: My intension is that other rules could be listed in this function, but `ctx` will be reused.
For example, a code like the following could be added in the future.

```
if what == 'module':
    _do_some_module_docstring_check(ctx)
```

So I think this is a good place to define `ctx`.

[the remark this answers is not in the document]
Comment (unnonouno) [reply docs/source/_docstring_check.py]: ok

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**